src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -332,12 +332,9 @@ internal bool CanReuseCompilationReferenceManager(CompilationOptions other)
             // In addition, the assembly name is determined based upon output kind. It is special for netmodules.
             // Can't reuse when file resolver or identity comparers change.
             // Can reuse even if StrongNameProvider changes. When resolving a cyclic reference only the simple name is considered, not the strong name.
-            return this.MetadataImportOptions == other.MetadataImportOptions
-                && this.ReferencesSupersedeLowerVersions == other.ReferencesSupersedeLowerVersions
-                && this.OutputKind.IsNetModule() == other.OutputKind.IsNetModule()
-                && object.Equals(this.XmlReferenceResolver, other.XmlReferenceResolver)
-                && object.Equals(this.MetadataReferenceResolver, other.MetadataReferenceResolver)
-                && object.Equals(this.AssemblyIdentityComparer, other.AssemblyIdentityComparer);
+            return this is { MetadataImportOptions: other.MetadataImportOptions, ReferencesSupersedeLowerVersions: other.ReferencesSupersedeLowerVersions } && this.OutputKind.IsNetModule() is other.OutputKind.IsNetModule() && object.Equals(this.XmlReferenceResolver, other.XmlReferenceResolver)
```

Comment:
>other.MetadataImportOptions [](start = 52, length = 27)

This is not a valid pattern (it isn't a constant expression)

Reply:
@alrz Action item here is to only use constant expressions in patterns

Reply:
📝 Constant checking is commented out for testing purposes. Should be enabled eventually.

Reply:
This appears to be fixed now.